dpnp/backend/kernels/dpnp_krnl_random.cpp

```diff
@@ -533,6 +533,292 @@ void dpnp_rng_negative_binomial_c(void* result, const double a, const double p,
     }
 }
 
+template <typename _DataType>
+void dpnp_rng_noncentral_chisquare_c(void* result, const _DataType df, const _DataType nonc, const size_t size)
+{
+    if (!size)
+    {
+        return;
+    }
+    _DataType* result1 = reinterpret_cast<_DataType*>(result);
+
+    const _DataType d_zero = _DataType(0.0);
+    const _DataType d_one = _DataType(1.0);
+    const _DataType d_two = _DataType(2.0);
+
+    if (dpnp_queue_is_cpu_c())
+    {
+        _DataType shape, loc;
+        size_t i;
```

Comment:
if you use this variable for loops only - use standard way to declare it inside loop boby